tests/ui/if_let_redundant_pattern_matching.stderr

```diff
@@ -24,5 +24,53 @@ error: redundant pattern matching, consider using `is_some()`
 27 |     if let Some(_) = Some(42) {
    |     -------^^^^^^^----------- help: try this: `if Some(42).is_some()`
 
-error: aborting due to 4 previous errors
+error: redundant pattern matching, consider using `is_ok()`
+  --> $DIR/if_let_redundant_pattern_matching.rs:47:9
+   |
+46 |     match Ok::<i32, i32>(42) {
+   |     ------------------------ help: try this: `if Ok::<i32, i32>(42).is_ok()`
```

Comment:
What is the exact span of this? If you want this to be machine applicable it needs to change

```rust
    match Ok::<i32, i32>(42) {
        Ok(_) => $1,
        Err(_) => $2,
    };
```

to

```rust
if Ok::<i32, i32>(42).is_ok() { $1 } else { $2 }
```

precisely

Reply:
@killercup Yeah! Good point! That would be a better and useful suggestion! Let me update it. Thanks!

Reply:
@killercup After thinking more, I think we don't need to check if the expression is assignment or not, the main purpose of this lint is to find unnecessary pattern matching if the match body are just booleans.

So I think the suggestion should be changed from

```
help: try this: `if Ok::<i32, i32>(42).is_ok()`
```

to

```
help: try this: `Ok::<i32, i32>(42).is_ok()`
```

I think it would be machine applicable because we just need to change `match Ok::<i32, i32>(42)` to `Ok::<i32, i32>(42).is_ok()`.

If I'm wrong, please correct me 🙇

Reply:
You need to replace the entire `match ... { ... }`, not just the part before the `{`.

Reply:
@CYBAI, ah, it was I who misunderstood what this lint does. Please file an issue for the case I suggested :)

My main concern was that the suggestion was not autofixable, since you replace too little of the original code. If you cover the entire match block, like Oli said, it should work.

Reply:
@oli-obk @killercup Understood! Let me try to figure out how to suggest the entire `match ... { ... }`; then I think it can be autofixable!

For the case @killercup suggested, I'll file one later 😃